### PR TITLE
Configurable mask for masked filters

### PR DESCRIPTION
**Pull request by timokroeger**
I adjusted the tests to compile, but I cannot run them unfortunately.

### CODE REVIEW

File: testsuite/tests/integration.rs [hunk 1]
```diff
@@ -131,10 +131,11 @@ mod tests {
     #[test]
     fn filter_mask32_std(state: &mut State) {
         let target_id = StandardId::new(42).unwrap();
+        let mask = StandardId::new(0x7FF).unwrap(); // Exact match required
```
Comment (jonas-schievink) [inline testsuite/tests/integration.rs:134]:
```suggestion
        let mask = StandardId::MAX; // Exact match required
```

(also below)

**Review by jonas-schievink — COMMENTED** [review]
I think we might want to make these separate constructors – it seems fairly common to create a `Mask32` that matches exactly one address for example, and having a constructor that does that without needing a `mask` is nice

**Comment by jonas-schievink** [comment]
Oh, also, feel free to add yourself to the crate authors – you did the base work for this crate anyways. I accidentally omitted you when instantiating the crate template.

**Comment by timokroeger** [comment]
First of all I don’t feel strongly about the design of the filter API. To move forward I can create the proposed changes as additional constructors.

Anyway I wanted to document my assumptions when I created the filter API of the original PR.

I had following common uses cases in mind which can be created with a constructor.
* Match exactly one StdID -> List16/List32
* Match exactly one ExtId -> List32
* StdId masked filters (only data frames) -> Mask16/Mask32
* ExtId masked filters (only data frames) -> Mask32

Special cases for masked filters which I’ve never seen used anywhere:
* Filters for remote frames (CANopen requires them for node-gaurding but even that has been replaced by heartbeat)
* Filters to accept both Std and Ext IDs

For the special cases I used modifer functions like `allow_remote()`, `remote_only()` and `allow_standard_and_extended()` (similar to the builder pattern) to refine the filter rule

**Comment by jonas-schievink** [comment]
Ah yeah, that makes sense. Let's leave the API as-is then.

**Pull request merged**